### ISSUE TITLE
What are the specifications we use? 

### ISSUE THREAD

**Issue by wiredsister** [verbosemode/syslogd-mirage#7]
Per conversation on IRC, which specs are we using and how much of those specs are implemented at present? Maybe we need to create a wiki/document that shows progress for others who want to jump in.

**Comment by verbosemode** [comment]
That's a great idea. Let's just start with wiki pages in this repo. We can move it to a Canopy Unikernel later :-) I also need to read through the IRC logs later on since I couldn't make it to the meeting.

**Comment by wiredsister** [comment]
Great. I can work on that :) 
Picking and choosing from this? --> http://www.rfc-base.org/rfc-5424.html

**Comment by wiredsister** [comment]
Which, basically links to the BSD spec which you have listed in the syslog-message repo (3164 I think)

**Comment by verbosemode** [comment]
Right, the initial idea was to implement RFC3164 with UDP as transport to get started, because that would be the simplest setup. I haven't read all the RFCs yet, but component wise there should be only a reporter (client/sender), a collector (server) and a relay (like a server, but just forwards msgs to one or more collectors). The implementation also should allow to change the transport protocol and use TLS over TCP instead of just UDP for example. 

RFC 5424 support would be also great, but it has UTF8 support and more fields, which will need a more sophisticated parser :-)

**Comment by wiredsister** [comment]
Adding in my blog post to this thread: 
http://www.gina.codes/ocaml/2016/06/06/syslog-a-tale-of-specifications.html
